packages/terminal-external/src/electron-node/terminal-external-service.ts

```diff
@@ -0,0 +1,196 @@
+
+    private static readonly CMD = 'cmd.exe';
+
+    private static DEFAULT_TERMINAL_WINDOWS: string;
+
+    async openTerminal(configuration: TerminalExternalConfiguration, cwd: URI): Promise<void> {
+        await this.spawnTerminal(configuration, FileUri.fsPath(cwd));
+    }
+
+    private async spawnTerminal(configuration: TerminalExternalConfiguration, cwd?: string): Promise<void> {
+        const terminalConfig = configuration['terminal.external.windowsExec'];
+        const exec = terminalConfig || WindowsTerminalExternalService.getDefaultTerminalWindows();
+
+        // Make the drive letter uppercase on Windows.
+        if (cwd && cwd[1] === ':') {
+            cwd = cwd[0].toUpperCase() + cwd.substr(1);
+        }
+
+        // cmder ignores the environment cwd and instead opts to always open in %USERPROFILE%
+        // unless otherwise specified.
+        const basename = path.basename(exec).toLowerCase();
+        if (basename === 'cmder' || basename === 'cmder.exe') {
+            cp.spawn(exec, cwd ? [cwd] : undefined);
+            return Promise.resolve(undefined);
+        }
+
+        const cmdArgs = ['/c', 'start', '/wait'];
+        // The "" argument is the window title. Without this, exec doesn't work when the path contains spaces.
+        if (exec.indexOf(' ') >= 0) {
+            cmdArgs.push('""');
+        }
+
+        cmdArgs.push(exec);
+
+        // Add starting directory parameter for Windows Terminal app.
+        if (basename === 'wt' || basename === 'wt.exe') {
+            cmdArgs.push('-d .');
+        }
+
+        return new Promise<void>(async (c, e) => {
+            const env = cwd ? { cwd: cwd } : undefined;
+            const command = await this.getWindowsShell();
+            const child = cp.spawn(command, cmdArgs, env);
+            child.on('error', e);
+            child.on('exit', () => c());
+        });
+    }
+
+    /**
+     * Get the default terminal app on Windows.
+     * (determine and initialize the variable if not found).
+     */
+    public static getDefaultTerminalWindows(): string {
+        if (!WindowsTerminalExternalService.DEFAULT_TERMINAL_WINDOWS) {
+            const isWoW64 = !!process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
+            WindowsTerminalExternalService.DEFAULT_TERMINAL_WINDOWS = `${process.env.windir ? process.env.windir : 'C:\\Windows'}\\
+            ${isWoW64 ? 'Sysnative' : 'System32'}\\cmd.exe`;
+        }
+        return WindowsTerminalExternalService.DEFAULT_TERMINAL_WINDOWS;
+    }
+
+    /**
+     * Find the Windows Shell process to start up (default to cmd.exe).
+     */
+    private async getWindowsShell(): Promise<string> {
+        const envVariables = await this.envVariablesServer.getVariables();
+        // Find the path to cmd.exe if possible (%compsec% environment variable).
+        const compsec = envVariables.find(e => e.name === 'compsec');
+        return compsec?.value || WindowsTerminalExternalService.CMD;
```

Comment:
This file is in `electron-node`, you don't need the `envVariableServer`, just use `process.env`.